packages/@aws-cdk/aws-ecs/lib/linux-parameters.ts

```diff
@@ -61,8 +97,22 @@ export class LinuxParameters extends Construct {
   constructor(scope: Construct, id: string, props: LinuxParametersProps = {}) {
     super(scope, id);
 
+    this.validateProps(props);
+
     this.sharedMemorySize = props.sharedMemorySize;
     this.initProcessEnabled = props.initProcessEnabled;
+    this.maxSwap = props.maxSwap;
+    this.swappiness = props.maxSwap ? props.swappiness : undefined;
+  }
+
+  private validateProps(props: LinuxParametersProps) {
+    if (props.maxSwap !== undefined && (!Number.isInteger(props.maxSwap) || props.maxSwap < 0)) {
+      throw new Error(`maxSwap: Must be a positive integer; received ${props.maxSwap}.`);
+    }
+
+    if (props.swappiness !== undefined && (!Number.isInteger(props.swappiness) || props.swappiness < 0 || props.swappiness > 100)) {
```

Comment:
Could `swappiness` be a Token? I believe we should check for this:
```suggestion
    if (!cdk.Token.isUnresolved(props.swappiness) && props.swappiness !== undefined && (!Number.isInteger(props.swappiness) || props.swappiness < 0 || props.swappiness > 100)) {
```